src/VirtualClient/VirtualClient.Actions.UnitTests/Hadoop/HadoopMetricsParserTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using global::VirtualClient;
+    using global::VirtualClient.Contracts;
+    using NUnit.Framework;
+
+    [TestFixture]
+    [Category("Unit")]
+    public class HadoopParserUnitTests
+    {
+        private string teragentResultRawText;
+        private string terasortResultRawText;
+        private HadoopMetricsParser teragenTestParser;
+        private HadoopMetricsParser terasortTestParser;
+        private IList<Metric> teragenMetrics;
+        private IList<Metric> terasortMetrics;
+
+        [SetUp]
+        public void Setup()
+        {
+            string workingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            string teragenOutputPath = Path.Combine(workingDirectory, @"Examples\Hadoop\HadoopTeragenExample.txt");
+            string terasortOutputPath = Path.Combine(workingDirectory, @"Examples\Hadoop\HadoopTerasortExample.txt");
+
+            this.teragentResultRawText = File.ReadAllText(teragenOutputPath);
+            this.terasortResultRawText = File.ReadAllText(terasortOutputPath);
+
+            this.teragenTestParser = new HadoopMetricsParser(this.teragentResultRawText);
+            this.terasortTestParser = new HadoopMetricsParser(this.terasortResultRawText);
+        }
+
+        [Test]
+        public void HadoopTerasortParserVerifySingleCoreResult()
+        {
+            this.teragenMetrics = this.teragenTestParser.Parse();
+            this.terasortMetrics = this.terasortTestParser.Parse();
+
+            this.teragenTestParser.JobCounters.PrintDataTableFormatted();
+            this.terasortTestParser.JobCounters.PrintDataTableFormatted();
+
+            Assert.AreEqual(4, this.teragenTestParser.JobCounters.Columns.Count);
+            Assert.AreEqual(4, this.terasortTestParser.JobCounters.Columns.Count);
+        }
+
+        [Test]
+        public void HadoopTeragenResultsParserCreatesTheExpectedMetricsFromResults()
+        {
```

Comment:
34 metrics or 54 metrics are a bit too much, is it possible to trim it down? to the <10 that matters? Like the total time, cpu time, i.e. the metrics that we can do regression comparison on

Reply:
The metrics include data such as the amount of bytes read and written during different sorting operations, along with memory snapshots. Conducting a comparative analysis for system performance would likely be beneficial. If these values are not extensively utilized in the future, we can make the adjustments.